SpatialGDK/Source/SpatialGDK/Private/SpatialView/OpList/ViewDeltaLegacyOpList.cpp

```diff
@@ -74,153 +45,108 @@ OpList GetOpListFromViewDelta(ViewDelta Delta)
-		Op.op.component_update.update = Worker_ComponentUpdate{
-			nullptr, Update.Update.GetComponentId(),
-			Update.Update.GetUnderlying(), nullptr
-		};
-		Ops.Push(Op);
-	}
-
-	for (const EntityComponentId& Id : Delta.GetComponentsRemoved())
-	{
-		Worker_Op Op = {};
-		Op.op_type = WORKER_OP_TYPE_REMOVE_COMPONENT;
-		Op.op.remove_component.entity_id = Id.EntityId;
-		Op.op.remove_component.component_id = Id.ComponentId;
-		Ops.Push(Op);
-	}
+		Worker_Op StartCriticalSection = {};
+		StartCriticalSection.op_type = WORKER_OP_TYPE_CRITICAL_SECTION;
+		StartCriticalSection.op.critical_section.in_critical_section = 1;
+		Ops.Add(StartCriticalSection);
 
-	for (const EntityComponentId& Id : Delta.GetAuthorityLostTemporarily())
-	{
-		Worker_Op Op = {};
-		Op.op_type = WORKER_OP_TYPE_AUTHORITY_CHANGE;
-		Op.op.authority_change.entity_id = Id.EntityId;
-		Op.op.authority_change.component_id = Id.ComponentId;
-		Op.op.authority_change.authority = WORKER_AUTHORITY_AUTHORITATIVE;
-		Ops.Push(Op);
-	}
+		if (Entity.bAdded)
+		{
+			Worker_Op Op = {};
+			Op.op_type = WORKER_OP_TYPE_ADD_ENTITY;
+			Op.op.add_entity.entity_id = Entity.EntityId;
+			Ops.Push(Op);
+		}
 
-	for (const EntityComponentId& Id : Delta.GetAuthorityGained())
-	{
-		Worker_Op Op = {};
-		Op.op_type = WORKER_OP_TYPE_AUTHORITY_CHANGE;
-		Op.op.authority_change.entity_id = Id.EntityId;
-		Op.op.authority_change.component_id = Id.ComponentId;
-		Op.op.authority_change.authority = WORKER_AUTHORITY_AUTHORITATIVE;
-		Ops.Push(Op);
-	}
+		for (const ComponentChange& Change : Entity.ComponentChanges)
+		{
+			if (Change.Type == ComponentChange::ADD)
+			{
+				Worker_Op Op = {};
+				Op.op_type = WORKER_OP_TYPE_ADD_COMPONENT;
+				Op.op.add_component.entity_id = Entity.EntityId;
+				Op.op.add_component.data = Worker_ComponentData{ nullptr, Change.ComponentId, Change.Data, nullptr };
+				Ops.Push(Op);
+			}
+		}
 
-	for (const Worker_EntityId& Id : Delta.GetEntitiesRemoved())
-	{
-		Worker_Op Op = {};
-		Op.op_type = WORKER_OP_TYPE_REMOVE_ENTITY;
-		Op.op.remove_entity.entity_id = Id;
-		Ops.Push(Op);
-	}
+		for (const AuthorityChange& Change : Entity.AuthorityChanges)
+		{
+			if (Change.Type == AuthorityChange::AUTHORITY_LOST || Change.Type == AuthorityChange::AUTHORITY_LOST_TEMPORARILY)
+			{
+				Worker_Op Op = {};
+				Op.op_type = WORKER_OP_TYPE_AUTHORITY_CHANGE;
+				Op.op.authority_change.entity_id = Entity.EntityId;
+				Op.op.authority_change.component_id = Change.ComponentId;
+				Op.op.authority_change.authority = WORKER_AUTHORITY_NOT_AUTHORITATIVE;
+				Ops.Push(Op);
+			}
+		}
 
-	// Sort the entity ops by entity ID and surround each set of entity ops with a critical section.
-	Algo::StableSort(Ops, [](const Worker_Op& Lhs, const Worker_Op& Rhs)
-	{
-		return GetEntityIdFromOp(Lhs) < GetEntityIdFromOp(Rhs);
-	});
+		for (const ComponentChange& Change : Entity.ComponentChanges)
+		{
+			if (Change.Type == ComponentChange::COMPLETE_UPDATE)
```

Comment:
We can't put them with the actual adds because a real add can't come after authority is gained, where as this one can (as it can come after authority is lost).

Reply:
So they go with updates because they are state changes that can only be received while you don't have authority.